docs/patterns/methodoverrides.rst

```diff
@@ -29,7 +29,7 @@ This can easily be accomplished with an HTTP middleware::
         def __call__(self, environ, start_response):
             method = environ.get('HTTP_X_HTTP_METHOD_OVERRIDE', '').upper()
             if method in self.allowed_methods:
-                method = method.encode('ascii', 'replace')
+                method = method.encode('ascii', 'replace')# Remove this line if you are using Python3. 
```

Comment:
Could you add an explanation of why you recommend removing that line so that people can get a better understanding?

Reply:
I tried use Method override pattner and it was crashed when method.encode('ascii', 'replace') because the next method needs string value and not byte value.

Reply:
with this line, the app crashs.

Reply:
@grahamlyons Please execute test in python3 that enter in this conditional.

[Python Method encode 3](https://docs.python.org/3/howto/unicode.html#converting-to-bytes) -> convert to Bytes.

[Python Method encode 2](https://docs.python.org/2/howto/unicode.html#the-unicode-type)  -> convert to String.

Reply:
I tested this example in Python 3 and  it worked fine.
Normally, the variable `method` should be `str` type. I don't know how you made it into `bytes`. So can you give more details?

Reply:
Now the description of this problem is very clear. I think you'd better post the stack traceback of the error in the issue #2981. This will help you describe the problem more accurately.

Just as an example, I think you can remove this line of code directly, and it will work fine under python 2 as well.  As to which solution is better, you can consult @davidism!

Reply:
@garenchan I apologize for my mistakes.

Reply:
That's alright.